src/lib/provable/int.ts

```diff
@@ -532,6 +534,68 @@ class UInt64 extends CircuitValue {
+   *
+   * If you use the optional `length` argument, proves that the UInt64 element fits in `length` bits.
+   * The `length` has to be between 0 and 64 and the method throws if it isn't.
+   *
+   * **Warning**: The cost of this operation in a zk proof depends on the `length` you specify,
+   * which by default is 64 bits. Prefer to pass a smaller `length` if possible.
+   *
+   * @param length - the number of bits to fit the element. If the element does not fit in `length` bits, the functions throws an error.
+   *
+   * @return An array of {@link Bool} element representing little endian binary representation of this {@link UInt64}.
+   */
+  toBits(length: number = 64) {
+    checkBitLength('UInt64.toBits()', length, 64);
+    if (this.isConstant()) {
+      let bits = BinableFp.toBits(this.toBigInt());
+      if (bits.slice(length).some((bit) => bit))
+        throw Error(`UInt64.toBits(): ${this} does not fit in ${length} bits`);
+      return bits.slice(0, length).map((b) => new Bool(b));
+    }
+    let bits = Provable.witness(Provable.Array(Bool, length), () => {
+      let f = this.toBigInt();
+      return Array.from({ length }, (_, k) => new Bool(!!((f >> BigInt(k)) & 0x1n)));
+    });
+    UInt64.fromBits(bits).assertEquals(this);
+    return bits;
+  }
+
+  /**
+   * Convert a bit array into a {@link UInt64} element using [little endian binary representation](https://en.wikipedia.org/wiki/Endianness)
+   *
+   * The method throws if the given bits do not fit in a single UInt64 element. In this case, no more than 64 bits are allowed.
+   *
+   * **Important**: If the given `bits` array is an array of `booleans` or {@link Bool} elements that all are `constant`,
+   *  the resulting {@link UInt64} element will be a constant as well. Or else, if the given array is a mixture of constants and variables of {@link Bool} type,
+   *  the resulting {@link UInt64} will be a variable as well.
+   *
+   * @param bits - An array of {@link Bool} or `boolean` type.
+   *
+   * @return A {@link UInt64} element matching the [little endian binary representation](https://en.wikipedia.org/wiki/Endianness) of the given `bits` array.
+   */
+  static fromBits(bits: (Bool | boolean)[]) {
+    const length = bits.length;
+    checkBitLength('UInt64.fromBits()', length, 64);
+    if (bits.every((b) => typeof b === 'boolean' || b.toField().isConstant())) {
+      let bits_ = bits
+        .map((b) => (typeof b === 'boolean' ? b : b.toBoolean()))
+        .concat(Array(Fp.sizeInBits - length).fill(false));
+      return new UInt64(BinableFp.fromBits(bits_));
+    }
+    return UInt64.Unsafe.fromField(
+      bits
+        .map((b) => new Bool(b))
+        .reduce((acc, bit, idx) => {
+          const shift = 1n << BigInt(idx);
+          return acc.add(bit.toField().mul(shift));
+        }, Field.from(0))
+        .seal()
+    );
```

Comment:
just use `Field.fromBits()`
```suggestion
    return UInt64.Unsafe.fromField(Field.fromBits(bits));
```

Reply:
You're right—I need to be more attentive and get more familiar with the library